### ISSUE TITLE
Raspberry Pi version please!

### ISSUE THREAD

**Issue by JeniT** [cgutteridge/geocraft#4]
This is awesome, Chris.

I tried to get it working on a Raspberry Pi 2 and generated a small world OK, but the save format looks to be different for minecraft-pi than what's being generated. I imagine this is because minecraft-pi hasn't been updated in years and the save format has moved on.

If you can get it working I'll happily write some documentation. Also happy to test if you don't have a Pi handy.

**Comment by cgutteridge** [comment]
I have done a bit of research and it seems that it's an entirely different save file format; http://minecraft.gamepedia.com/Old and only 256x256 so I really don't feel it's worth the effort. Sorry Jeni.

**Comment by JeniT** [comment]
OK. I thought it would be worthwhile because if the higher proportion of code clubs/schools that would have access to Raspberry Pis than Macs or full Linux boxes, if kids are your target for this as you implied in the video.

But I appreciate that the fact it's an entirely different save format makes it a huge piece of work. Probably better for me to lobby for Minecraft Pi to be updated to use the new save format!

**Comment by jargonautical** [comment]
I'd like to do this on a Minecraft server running on a Pi - not Minecraft Pi but the full version, accessed through another machine - but I'm stuck at the perl modules bit. Any suggestions as to where I look for inspiration?

**Comment by cgutteridge** [comment]
The only module it needs is ImageMagick which it uses to get the colour of a pixel out of a PNG. I'm tempted to try to find an alternative which would simplify installation no end...

**Comment by cgutteridge** [comment]
Hi, I've done quite a bit of work on this tonight and it no longer has the dependency on ImageMagick. I wrote a cheap and cheerful PNG parser as part of the tool. You'll still need Archive::Zip but that's much easier to install.

Let me know how you get on.

**Comment by jargonautical** [comment]
Thanks Chris - much appreciated. Just working on it now and it seems to be generating the world ok but I've yet to explore it. Will report back as soon as I have a clue how I'm doing :)

**Comment by cgutteridge** [comment]
Thanks. I'd love to hear about how you get on.

On 30/01/2016 17:35, Lucy Knight wrote:

> Thanks Chris - much appreciated. Just working on it now and it seems 
> to be generating the world ok but I've yet to explore it. Will report 
> back as soon as I have a clue how I'm doing :)
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/cgutteridge/geocraft/issues/4#issuecomment-177258176.Web 
> Bug from 
> https://github.com/notifications/beacon/AAfK64KqvOu64i76phQU7-ugJFWA7k0uks5pfOv5gaJpZM4G7f-B.gif

## 

Christopher Gutteridge -- http://users.ecs.soton.ac.uk/cjg

University of Southampton Open Data Service: http://data.southampton.ac.uk/
You should read the ECS Web Team blog: http://blogs.ecs.soton.ac.uk/webteam/

**Comment by jargonautical** [comment]
I'm stuck somewhere - being new to running a Minecraft server, I'm not sure where I'm going wrong.
I have a functional (if laggy and crashy) server ... I have a generated world called Highfield based on the  SO postcode, in the 'saves' folder - but when I join the server I just get a standard generated wooded hills biome new world. I'm likely missing something obvious ... Is there a special trick to forcing the server to use the generated world?

**Comment by cgutteridge** [comment]
Probably need to copy the file into the _actual_ Minecraft saves folder 
... do you see it as an option on single player?

On 31/01/2016 11:41, Lucy Knight wrote:

> I'm stuck somewhere - being new to running a Minecraft server, I'm not 
> sure where I'm going wrong.
> I have a functional (if laggy and crashy) server ... I have a 
> generated world called Highfield based on the SO postcode, in the 
> 'saves' folder - but when I join the server I just get a standard 
> generated wooded hills biome new world. I'm likely missing something 
> obvious ... Is there a special trick to forcing the server to use the 
> generated world?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/cgutteridge/geocraft/issues/4#issuecomment-177474563.Web 
> Bug from 
> https://github.com/notifications/beacon/AAfK6woxO6kpETLp-FsUyCTXDiBn7Wcpks5pfepxgaJpZM4G7f-B.gif

## 

Christopher Gutteridge -- http://users.ecs.soton.ac.uk/cjg

University of Southampton Open Data Service: http://data.southampton.ac.uk/
You should read the ECS Web Team blog: http://blogs.ecs.soton.ac.uk/webteam/

**Comment by jargonautical** [comment]
No, it doesn't show up in single player - I tried creating a new world there and forcing it to take the name of the generated world, but that didn't work. I'll try placing the file as you suggest and go from there. I assume the Minecraft saves folder is on the client side? Sorry for the amateur questions :(

**Comment by cgutteridge** [comment]
Here's a tip to find it; in the game use "open resourcepacks folder" 
feature... the resource pack folder is in the same parent as the saves 
folder...

On 31/01/2016 13:03, Lucy Knight wrote:

> No, it doesn't show up in single player - I tried creating a new world 
> there and forcing it to take the name of the generated world, but that 
> didn't work. I'll try placing the file as you suggest and go from 
> there. I assume the Minecraft saves folder is on the client side? 
> Sorry for the amateur questions :(
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/cgutteridge/geocraft/issues/4#issuecomment-177496565.Web 
> Bug from 
> https://github.com/notifications/beacon/AAfK661eEnuihncVaC65treEnwyVveTQks5pff2SgaJpZM4G7f-B.gif

## 

Christopher Gutteridge -- http://users.ecs.soton.ac.uk/cjg

University of Southampton Open Data Service: http://data.southampton.ac.uk/
You should read the ECS Web Team blog: http://blogs.ecs.soton.ac.uk/webteam/

**Comment by jargonautical** [comment]
OK, found that ok :) I've got stuff happening but all it will generate for me is a 1000-tile square of bedrock! this happens every time at this point, whichever postcode i try ...
-500..500,-439
-500..500,-438
-500..500,-437
-500..500,-436
-500..500,-435
failed to open /Users/lucyknight/Sites/geocraft-master/saves/Highfield2/region/r.-1.-1.mca: No such file or directory at /Users/lucyknight/Sites/geocraft-master/lib/Minecraft/Region.pm line 273.
SAVING REGION r.-1.-1.mca ..

**Comment by cgutteridge** [comment]
Hmmm. Are you local to Highfield? If so it might be easier to solve in 
person!

On 03/02/2016 21:12, Lucy Knight wrote:

> OK, found that ok :) I've got stuff happening but all it will generate 
> for me is a 1000-tile square of bedrock! this happens every time at 
> this point, whichever postcode i try ...
> -500..500,-439
> -500..500,-438
> -500..500,-437
> -500..500,-436
> -500..500,-435
> failed to open 
> /Users/lucyknight/Sites/geocraft-master/saves/Highfield2/region/r.-1.-1.mca: 
> No such file or directory at 
> /Users/lucyknight/Sites/geocraft-master/lib/Minecraft/Region.pm line 273.
> SAVING REGION r.-1.-1.mca ..
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/cgutteridge/geocraft/issues/4#issuecomment-179470624.Web 
> Bug from 
> https://github.com/notifications/beacon/AAfK63oHWfdiFy0IZFfPcD6zx908avOaks5pgmSjgaJpZM4G7f-B.gif

## 

Christopher Gutteridge -- http://users.ecs.soton.ac.uk/cjg

University of Southampton Open Data Service: http://data.southampton.ac.uk/
You should read the ECS Web Team blog: http://blogs.ecs.soton.ac.uk/webteam/

**Comment by jargonautical** [comment]
Sadly not - I'm all the way down in Torbay! though we're hoping to use this for a hack somewhere in Exeter in a couple of months, so if I haven't got it working by then we'll bribe you with cake to come along and see what you can do?

**Comment by jargonautical** [comment]
Is it remotely possible this is an internet connection issue? My wifi has a tendency to blip every once in a while, I could get a hard wire connection and see if that makes a difference.

**Comment by jargonautical** [comment]
Success!
Managed to generate the Highfield file, daughter is exploring it now :) I think the wifi issue might have been the last hurdle, I got an Ethernet adapter for the MacBook today and made sure I had solid connection the whole time it was generating the world. 
Next step - maybe I should write the definitive n00bs guide ...

**Comment by cgutteridge** [comment]
Great news! The system doesn't fail when it misses data, but maybe I 
should chose a block as an "error" block if there's no map tile loaded? 
Or make a report a the end?

On 04/02/2016 21:24, Lucy Knight wrote:

> Success!
> Managed to generate the Highfield file, daughter is exploring it now 
> :) I think the wifi issue might have been the last hurdle, I got an 
> Ethernet adapter for the MacBook today and made sure I had solid 
> connection the whole time it was generating the world.
> Next step - maybe I should write the definitive n00bs guide ...
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/cgutteridge/geocraft/issues/4#issuecomment-180059350.Web 
> Bug from 
> https://github.com/notifications/beacon/AAfK64iiUTDAHl-2AcMmvWfOjkVyv03oks5pg7kYgaJpZM4G7f-B.gif

## 

Christopher Gutteridge -- http://users.ecs.soton.ac.uk/cjg

University of Southampton Open Data Service: http://data.southampton.ac.uk/
You should read the ECS Web Team blog: http://blogs.ecs.soton.ac.uk/webteam/

**Comment by jargonautical** [comment]
Both would be helpful. Anything that flags the errors without stopping the whole process would be a bonus, helps people like me navigate through it
